packages/ag-charts-enterprise/src/box-plot/boxPlotSeries.ts

```diff
@@ -0,0 +1,319 @@
+    fill: string = '#c16068';
+
+    @Validate(OPT_COLOR_STRING)
+    stroke: string = '#333';
+
+    @Validate(NUMBER(0))
+    strokeWidth: number = 3;
+
+    tooltip = new SeriesTooltip<AgCartesianSeriesTooltipRendererParams>();
+
+    getBandScalePadding() {
+        return { inner: 0.2, outer: 0.3 };
+    }
+
+    async createNodeData() {
+        const { visible, dataModel } = this;
+
+        const xAxis = this.axes[ChartAxisDirection.X];
+        const yAxis = this.axes[ChartAxisDirection.Y];
+
+        if (!(dataModel && visible && xAxis && yAxis)) {
+            return [];
+        }
+
+        const { yKey = '', fill, stroke, strokeWidth, processedData } = this;
+
+        const context: _ModuleSupport.SeriesNodeDataContext<BoxPlotNodeDatum> = {
+            itemId: yKey,
+            nodeData: [],
+            labelData: [],
+        };
+
+        const yValueIndex = dataModel.resolveProcessedDataIndexById(this, `yValue`).index;
+        const minValueIndex = dataModel.resolveProcessedDataIndexById(this, `minValue`).index;
+        const q1ValueIndex = dataModel.resolveProcessedDataIndexById(this, `q1Value`).index;
+        const medianValueIndex = dataModel.resolveProcessedDataIndexById(this, `medianValue`).index;
+        const q3ValueIndex = dataModel.resolveProcessedDataIndexById(this, `q3Value`).index;
+        const maxValueIndex = dataModel.resolveProcessedDataIndexById(this, `maxValue`).index;
+
+        processedData?.data.forEach(({ keys, datum: seriesDatum, values }, dataIndex) => {
+            const yValue = keys[yValueIndex];
+            const minValue = values[minValueIndex];
+            const q1Value = values[q1ValueIndex];
+            const medianValue = values[medianValueIndex];
+            const q3Value = values[q3ValueIndex];
+            const maxValue = values[maxValueIndex];
+
+            const nodeData: BoxPlotNodeDatum = {
+                index: dataIndex,
+                datum: seriesDatum,
+                series: this,
+                itemId: yKey,
+                xKey: '',
+                xValue: 0,
+                yKey,
+                fill,
+                stroke,
+                strokeWidth,
+                bandwidth: yAxis.scale.bandwidth ?? 0,
+                yValue: Math.round(yAxis.scale.convert(yValue, { strict: false })),
+                minValue: Math.round(xAxis.scale.convert(minValue, { strict: false })),
+                q1Value: Math.round(xAxis.scale.convert(q1Value, { strict: false })),
+                medianValue: Math.round(xAxis.scale.convert(medianValue, { strict: false })),
+                q3Value: Math.round(xAxis.scale.convert(q3Value, { strict: false })),
+                maxValue: Math.round(xAxis.scale.convert(maxValue, { strict: false })),
+            };
+
+            context.nodeData.push(nodeData);
+        });
+
+        return [context];
+    }
+
+    async processData(dataController: _ModuleSupport.DataController): Promise<void> {
```

Comment:
Nit: Could you move this and `getDomain` to appear above `createNodeData` - this is the order we use in all other series implementations, and it logically follows the processing order?